ast-nodes/index-access.dart

```diff
@@ -21,13 +24,35 @@ class IndexAccess implements ModifiablePrimary {
 
   IndexAccess(this.index, this.object);
 
-  // TODO: implement .parse()
+  factory IndexAccess.parse(Iterable<Token> tokens) {
```

Comment:
You may hate me, rightfully so, but now I'm thinking that it's best to just have a single `.parse` in the `modifiable-primary`, which would create `IndexAccess` and `FieldAccess` as needed. That's because we may have to deal with infinitely complex nesting like `a.b[c.d[e][f]][g.a].k`.

Reply:
So what exactly do you mean? Get rid of parsing function?

Reply:
Yes, get rid of it here and move it to `ModifiablePrimary`